arrow-select/src/take.rs

```diff
@@ -223,6 +223,22 @@ fn take_impl<IndexType: ArrowPrimitiveType>(
                 Ok(new_null_array(&DataType::Null, indices.len()))
             }
         }
+        DataType::Union(fields, UnionMode::Sparse) => {
+            let mut field_type_ids = Vec::with_capacity(fields.len());
+            let mut children = Vec::with_capacity(fields.len());
+            let values = values.as_any().downcast_ref::<UnionArray>().unwrap();
+            let type_ids = Int8Array::try_new(values.type_ids().clone(), values.nulls().cloned())?;
+            let type_ids = take_impl(&type_ids, indices)?.to_data().buffer(0).into();
```

Comment:
```suggestion
            let type_ids = take_native(values.type_ids(), indices);
```
UnionArray can't have nulls at this level, so this can be simplified